File: solutions/213.HouseRobber2/lina.py
class Solution(object):
    def rob(self, nums):

        return max(nums[0], self.helper(nums[1:]), self.helper(nums[:-1]))

    def helper(self, nums):
            prevMax, newMax = 0, 0

            for n in nums:
                newRob = max(prevMax + n, newMax)
                prevMax = newMax
                newMax = newRob
            return newMax
        
        

File: solutions/647.PalindromicSubstrings/lina.java
class Solution {
    public int output = 0;
    public int n = 0;
    public int countSubstrings(String s) {
        
        n = s.length();

        for(int i = 0; i < n; i++)
        {
            helper(i,i,s);
            helper(i,i+1,s);
        }

        return output;
    }
    public void helper(int i, int j, String s)
    {
        while(i > 0 && i < n && j > 0 && j < n && s[i] == s[j])
        {
            output += 1;
            i--;
            j++;
        }
    }
}